Library/Homebrew/livecheck/strategy.rb

```diff
@@ -119,25 +120,68 @@ def self.page_headers(url)
 
       # Fetches the content at the URL and returns a hash containing the
       # content and, if there are any redirections, the final URL.
+      # If `curl` encounters an error, the hash will contain a `:messages`
+      # array with the error message instead.
       #
       # @param url [String] the URL of the content to check
       # @return [Hash]
       sig { params(url: String).returns(T::Hash[Symbol, T.untyped]) }
       def self.page_content(url)
         original_url = url
 
-        # Manually handling `URI#open` redirections allows us to detect the
-        # resolved URL while also supporting HTTPS to HTTP redirections (which
-        # are normally forbidden by `OpenURI`).
-        begin
-          content = URI.parse(url).open(redirect: false, &:read)
-        rescue OpenURI::HTTPRedirect => e
-          url = e.uri.to_s
-          retry
+        args = curl_args(
+          "--compressed",
+          # Include HTTP response headers in output, so we can identify the
+          # final URL after any redirections
+          "--include",
+          # Follow redirections to handle mirrors, relocations, etc.
+          "--location",
+          # cURL's default timeout can be up to two minutes, so we need to
+          # set our own timeout settings to avoid a lengthy wait
+          "--connect-timeout", "10",
+          "--max-time", "15"
+        )
+
+        stdout, stderr, status = curl_with_workarounds(
+          *args, url,
+          print_stdout: false, print_stderr: false,
+          debug: false, verbose: false,
+          user_agent: :default, timeout: 20,
+          retry: false
+        )
+
+        unless status.success?
+          /^(?<error_msg>curl: \(\d+\) .+)/ =~ stderr
+          return { messages: [error_msg] } if error_msg.present?
```

Comment:
This won't return if there's no `error_msg`. Is that intentional?

Reply:
Good catch. This should always return when `status.success?` isn't true and I've added a commit to address this. It seems unlikely that we'll ever see the fallback `cURL failed without an error` message (if `curl` doesn't provide an error message) but it's there just in case.

If you have any feedback on what the fallback error message should say, I'm open to other suggestions.